staging/src/k8s.io/client-go/tools/watch/retrywatcher_test.go

```diff
@@ -0,0 +1,509 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watch
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"reflect"
+	"strconv"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/klog"
+)
+
+func init() {
+	// Enable klog which is used in dependencies
+	klog.InitFlags(nil)
+	flag.Set("logtostderr", "true")
+	flag.Set("v", "9")
+}
+
+type testObject struct {
+	resourceVersion string
+}
+
+func (o testObject) GetObjectKind() schema.ObjectKind { return schema.EmptyObjectKind }
+func (o testObject) DeepCopyObject() runtime.Object   { return o }
+func (o testObject) GetResourceVersion() string       { return o.resourceVersion }
+
+func withCounter(f WatcherFunc) (*uint32, func(sinceResourceVersion string) (watch.Interface, error)) {
+	var counter uint32
+	wrapped := func(sinceResourceVersion string) (watch.Interface, error) {
+		atomic.AddUint32(&counter, 1)
+		return f(sinceResourceVersion)
+	}
+
+	return &counter, wrapped
+}
+
+func makeTestEvent(rv int) watch.Event {
+	return watch.Event{
+		Type: watch.Added,
+		Object: testObject{
+			resourceVersion: fmt.Sprintf("%d", rv),
+		},
+	}
+}
+
+func AtoiOrDie(s string) int {
+	i, err := strconv.Atoi(s)
```

Comment:
clients should never assume the resource version is an int... the only valid test of a resource version is an equality check

from https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#metadata:

> resourceVersion: a string that identifies the internal version of this object that can be used by clients to determine when objects have changed. This value MUST be treated as opaque by clients and passed unmodified back to the server. Clients should not assume that the resource version has meaning across namespaces, different kinds of resources, or different servers.

Reply:
oh... just realized this is in a test file... in that case, this is acting as the server, so interpreting it is ok